File: python/ParkingLot.py
# /*Saya Rakha Dhifiargo Hariadi
#  NIM 2209489 mengerjakan soal 
#  Latihan praktikum 4 dalam mata
#  kuliah DPBO
#  untuk keberkahanNya maka saya tidak
#  melakukan kecurangan seperti 
#  yang telah dispesifikasikan. Aamiin.*/ 
from Vehicle import Vehicle 
 
class ParkingLot:
    
    __namaParkingLot = "" # atribut untuk parking lot
    __luasParkingLot = ""
    __kapasitas = 0
    __jumlahKendaraanNow = 0
    __daftarKendaraan = []

    # constructor with parameters
    def __init__(self, namaParkingLot="", luasParkingLot=0, kapasitas=0, jumlahKendaraanNow=0, daftarKendaraan=[]):
        self.__namaParkingLot = namaParkingLot
        self.__luasParkingLot = luasParkingLot
        self.__kapasitas = kapasitas
        self.__jumlahKendaraanNow = jumlahKendaraanNow
        self.__daftarKendaraan = daftarKendaraan

    # setter
    def setNamaParkingLot(self, namaParkingLot):
        self.__namaParkingLot = namaParkingLot

    def setLuasParkingLot(self, luasParkingLot):
        self.__luasParkingLot = luasParkingLot

    def setKapasitas(self, kapasitas):
        self.__kapasitas = kapasitas

    def setJumlahKendaraanNow(self, jumlahKendaraanNow):
        self.__jumlahKendaraanNow = jumlahKendaraanNow

    def setDaftarKendaraan(self, daftarKendaraan):
        self.__daftarKendaraan = daftarKendaraan

    # getter
    def getNamaParkingLot(self):
        return self.__namaParkingLot

    def getLuasParkingLot(self):
        return self.__luasParkingLot

    def getKapasitas(self):
        return self.__kapasitas

    def getJumlahKendaraanNow(self):
        return self.__jumlahKendaraanNow

    def getDaftarKendaraan(self):
        return self.__daftarKendaraan
    
    # method untuk menambah satu kendaraan ke array
    def addKendaraan(self, kendaraan):
        self.__daftarKendaraan.append(kendaraan)
        self.__jumlahKendaraanNow = self.__jumlahKendaraanNow + 1
    

File: cpp/ParkingLot.cpp
#pragma once
#include <bits/stdc++.h>
#include "Vehicle.cpp" // include the Vehicle class file

using namespace std;

class ParkingLot {
private:
    string namaParkingLot;   // attribute for parking lot
    int luasParkingLot;
    int kapasitas;
    int jumlahKendaraanNow;
    vector<Vehicle*> daftarKendaraan;

public:

    // konstruktor korong
    ParkingLot()
    {

    }

    // constructor with parameters
    ParkingLot(string namaParkingLot = "", int luasParkingLot = 0, int kapasitas = 0, int jumlahKendaraanNow = 0, vector<Vehicle*> daftarKendaraan = {})
    {
        this->namaParkingLot = namaParkingLot;
        this->luasParkingLot = luasParkingLot;
        this->kapasitas = kapasitas;
        this->jumlahKendaraanNow = jumlahKendaraanNow;
        this->daftarKendaraan = daftarKendaraan;
    }

    // setter for each attribute
    void setNamaParkingLot(string namaParkingLot) 
    {
        this->namaParkingLot = namaParkingLot;
    }

    void setLuasParkingLot(int luasParkingLot) 
    {
        this->luasParkingLot = luasParkingLot;
    }

    void setKapasitas(int kapasitas) 
    {
        this->kapasitas = kapasitas;
    }

    void setJumlahKendaraanNow(int jumlahKendaraanNow) 
    {
        this->jumlahKendaraanNow = jumlahKendaraanNow;
    }

    void setDaftarKendaraan(vector<Vehicle*> daftarKendaraan) 
    {
        this->daftarKendaraan = daftarKendaraan;
    }

    // getter for each attribute
    string getNamaParkingLot() 
    {
        return this->namaParkingLot;
    }

    int getLuasParkingLot() 
    {
        return this->luasParkingLot;
    }

    int getKapasitas() 
    {
        return this->kapasitas;
    }

    int getJumlahKendaraanNow() 
    {
        return this->jumlahKendaraanNow;
    }

    vector<Vehicle*> getDaftarKendaraan() 
    {
        return this->daftarKendaraan;
    }

    // method to add a vehicle to the vector
    void addKendaraan(Vehicle* kendaraan) 
    {
        this->daftarKendaraan.push_back(kendaraan);
        this->jumlahKendaraanNow = this->jumlahKendaraanNow + 1;
    }

    ~ParkingLot() // destruktor
    {
        
    }
};